FSI/flap_perp/OpenFOAM-FEniCS/Solid/perp-flap.py

```diff
@@ -63,176 +62,188 @@ def Neumann_Boundary(x, on_boundary):
 v_n = Function(V)
 a_n = Function(V)
 
+f_N_function = interpolate(Expression(("1", "0"), degree=1), V)
+u_function = interpolate(Expression(("0", "0"), degree=1), V)
 
-f_N_function = interpolate(Expression(("1","0"), degree=1), V)
-u_function = interpolate(Expression(("0","0"), degree=1), V)
-
-coupling_boundary = AutoSubDomain(Neumann_Boundary)
-
-# create subdomain that resembles the 
-
-## get the adapter ready
-
-#read fenics-adapter json-config-file)
+coupling_boundary = AutoSubDomain(neumann_boundary)
 
 adapter_config_filename = "precice-adapter-config-fsi-s.json"
-           
+
 precice = Adapter(adapter_config_filename)
 
-clamped_boundary_domain=AutoSubDomain(clamped_boundary)
-precice_dt = precice.initialize(coupling_subdomain=coupling_boundary, 
-                            mesh=mesh,
-                            read_field=f_N_function,
-                            write_field=u_function,
-                            u_n=u_n,
-                            dimension=dim,
-                            dirichlet_boundary=clamped_boundary_domain)
+clamped_boundary_domain = AutoSubDomain(clamped_boundary)
+force_boundary = AutoSubDomain(neumann_boundary)
 
-dt = Constant(precice_dt)
+# Initialize the coupling interface
+precice_dt = precice.initialize(coupling_boundary, mesh, f_N_function, V, dim)
 
-force_boundary = AutoSubDomain(Neumann_Boundary)
+fenics_dt = precice_dt  # if fenics_dt == precice_dt, no subcycling is applied
+# fenics_dt = 0.02  # if fenics_dt < precice_dt, subcycling is applied
+dt = Constant(np.min([precice_dt, fenics_dt]))
 
 # clamp the beam at the bottom
-bc = DirichletBC(V, Constant((0,0)), clamped_boundary)
-
-#alpha method parameters
+bc = DirichletBC(V, Constant((0, 0)), clamped_boundary)
 
+# alpha method parameters
 alpha_m = Constant(0.2)
 alpha_f = Constant(0.4)
-gamma   = Constant(0.5+alpha_f-alpha_m)
-beta    = Constant((gamma+0.5)**2/4.)
+gamma = Constant(0.5 + alpha_f - alpha_m)
+beta = Constant((gamma + 0.5) ** 2 / 4.)
```

Comment:
```suggestion
beta = Constant((gamma + 0.5)**2 / 4.)
```

Whitespace around `**` looks a bit odd.